react/__tests__/findSpecification.test.ts

```diff
@@ -0,0 +1,129 @@
+import findSpecificationLink from '../modules/findSpecification'
+import { PRODUCT_VARIABLES } from '../typings/types'
+
+const mockedSpecificationGroups = [
+  {
+    name: 'App Data',
+    specifications: [
+      {
+        name: 'google',
+        values: ['http://www.google.com/'],
+      },
+      {
+        name: 'mozila',
+        values: ['https://developer.mozilla.org/'],
+      },
+      {
+        name: '.linkedin',
+        values: ['https://www.linkedin.com/'],
+      },
+    ],
+  },
+  {
+    name: 'Storeframework',
+  },
+  {
+    name: '.groupTest',
+    specifications: [
+      {
+        name: '.test',
+        values: ['test'],
+      },
+      {
+        name: 'test',
+        values: ['test'],
+      },
+      {
+        name: 'test2',
+        values: [],
+      },
+    ],
+  },
+]
+
+describe('Specification Group Scenarios', () => {
+  it('Should find the link', () => {
+    const input = '{specificationGroups.App Data.specifications.google}'
```

Comment:
A different input is missing. Something like:

`/foobar?baz={specificationGroups.App Data.specifications.google}`

Reply:
query string should work ? My regex works if the the pattern was something like`{specificationGroups.|groupName|.specifications.|specificationName|}`. But I can update it.

Reply:
I updated the regex to consider only the value between `{}`